src/mono/mono/mini/aot-compiler.c

```diff
@@ -6588,12 +6588,9 @@ emit_and_reloc_code (MonoAotCompile *acfg, MonoMethod *method, guint8 *code, gui
 						direct_call_target = symbol;
 						patch_info->type = MONO_PATCH_INFO_NONE;
 					} else if ((m_class_get_image (patch_info->data.method->klass) == acfg->image) && !got_only && is_direct_callable (acfg, method, patch_info)) {
-#if 0
-						// FIXME: Currently not used. It fails as some callees require initialization.
 						MonoCompile *callee_cfg = (MonoCompile *)g_hash_table_lookup (acfg->method_to_cfg, cmethod);
 
-						// Don't compile inflated methods if we're doing dedup
-						if (acfg->aot_opts.dedup && !mono_aot_can_dedup (cmethod)) {
+						if (!m_class_has_cctor (method->klass) && mono_aot_can_specialize (callee_cfg->method)) {
 							char *name = mono_aot_get_mangled_method_name (cmethod);
```

Comment:
Looked into why this was not enabled before, and i don't think this is going to work. The callee method might use GOT slots etc. which are not initialized if the method is directly called. So before this can be enabled, the AOTed code needs to be changed to init itself, the same way the llvm compiled code does.

Reply:
Good point! Let's first try to enable method initialization during the invocation before proceeding with this and other direct call transformations.